### ISSUE TITLE
Possible false positive for Rule InsertWithoutColumns

### ISSUE THREAD

**Issue by phermsdorf** [felipebz/sonar-plsql#51]
Hi,

we often use code like

`insert into sometable values rec_sometable;`

where `rec_sometable` is defined as `sometable%rowtype`;

I think in this cases it's ok to not list every single column in the insert statement because it's strong typed.

What do you think about that?

PS: Bonus Use-Case:
```
for r_t in (select * from mytable where a=1) loop
    begin
        r_t.b := 7;
        insert into mytable values r_t;
    end;
end loop;
```
... probably harder to track ....

Thx! Bye Peter

**Comment by felipebz** [comment]
Hi @phermsdorf.

You're right, the plugin should ignore this situation. Right now I have a basic mechanism to identify the type of variables, I think I could improve it to check if it's an anchored datatype. ;-)

**Comment by felipebz** [comment]
Both cases were fixed in the current dev version (2.3.0-SNAPSHOT). :-)

**Comment by phermsdorf** [comment]
Thank you!
:+1:
